### ISSUE TITLE
Cruoromancer Infusion

### ISSUE THREAD

**Issue by Seinra** [Truinto/DarkCodex#287]
Hello, can you "tweak" the Cruoromancer "Commanding Infusion" so it adds the extra stats to ALL the Animate/Create summon type of necromantic spells? Because right now it only works with the "Animate Dead" spell, not even with the Animate Dead (lesser). I'm requesting you this because I saw that one of the new alchemist archetypes can actually add the extra stats to all the undeads it can summon.

**Comment by Truinto** [comment]
Reading the blueprint. There are a few weird properties that you might not expect reading the description. The ability creates an 30 feet aura around you and affects all friendly creatures which were summoned with animated dead.

So it does affect creatures summoned by allies. The order of summon does not matter. And the undead can lose/gain the buff by walking in and out of the 30 feet range.

**Comment by Seinra** [comment]
WTF? Hahahaha, I know Owlcat made some weird things, like with the bugs and unfinished feats for the Blood Kineticist, but writing a description for an ability which does not do that?
Ok, I'll have to request a "fix" to make the ability do what the description says, and then a "tweak", so it works with all the summoning spells from the Necromancy school.

**Comment by Truinto** [comment]
Increased the range to 100ft and the condition to friendly summoned undead. Write here if this does not work.

Btw, I find their solution practical. It solves a bunch of programming hurdles without extra work.

**Comment by Seinra** [comment]
Thanks! Now with your fix I can make the Cruoromancer + Agent of the Grave + Souldrinker I was thinking about.
I tested your fix casting Animate Dead/Create Undead spells, and it works on both spells, my companions can't summon undead creatures yet so I can't test if the infusion is still affecting them too. But I found an issue, Focused and Sickening infusions gives you a buff which disappears after you cast a spell, so you can use another infusion in the next turn, but this doesn't happen with the Commanding infusion, this buff doesn't disappear after casting a necromancy summon spell and it stays active even in the next turn buffing another summon spell, and preventing you from using another infusion.

**Comment by Truinto** [comment]
Too much text. I don't understand half of it.

Necromancer's Staff affects:
- DLC5_RingOfShadowStealingFeature
- SummonFrozenAbrikandiluAbility
- CreateUndeadGraveknight
- CreateUndeadLivingArmor
- CreateUndeadGreaterDevourer
- CreateUndeadGreaterDevourerSithhud
- CreateUndeadGreaterShadowGreater

**Comment by Seinra** [comment]
Ok, in few words:

https://pathfinderwrathoftherighteous.wiki.fextralife.com/Necromancer's+Staff
This says that Necromancer's Staff should work with Animate Dead spell, but it doesn't work with it.

About Cruoromancer's infusions, they should stop working after you cast a "compatible spell", this indeed happen when you cast a spell modified by Focused Infusion or Sickening Infusion, but Commanding Infusion keeps working after casting Animate Dead or Create Undead, this prevents you from using any of the other 2 infusions in the next turn.

**Comment by Truinto** [comment]
Better, but still difficult. Remember that I never used any of these and I am not going to boot up the game. Nor do I know the description of these abilities.

My first guess is you have a swift action ability and that's eating your actions, so you cannot use the other swift actions. The game does a poor job with action managment. Especially swift and move actions. If you think it's worthwhile, please explain to me what you want me to do.

I wrote a fix for the staff. It will simply affect all summoned undead (but only from the staff holder).

**Comment by Seinra** [comment]
Cruoromancer has three infusions that you can cast as a Swift Action, but you can only use one of the them per turn:

Lvl 1: Focused Infusion and Sickening Infusion. This two adds a effect to a single offensive necromancy spell.
Lvl 5: Commanding Infusion. This one adds stats to the undead summoned by a single necromancy spell.

⬆️ That is how the three infusions should work: you use the infusion to modify the effects of a necromancy spell, you cast the modified spell and the infusion effect is gone from your character, and in the next turn you are free to use another infusion.

But Commanding Infusion has a bug, when you use it in the 1st turn and then you animate/create an undead, and then you finish your turn, in the next turn you will notice that Commanding Infusion is still active, which means two things:

1) You can't use another infusion until the 3rd turn.
2) In the 2nd turn you can summon another undead added stats for the infusion from the 1st turn.

**Comment by Truinto** [comment]
Oh. Now I get it. Well, that not how the blueprint works at all. The first two infusions do get removed after being used. While Commanding just sticks for 1 round. I also just realize that the buff to the undead always vanish after this first round. This whole feature is completely wrong.

**Comment by Truinto** [comment]
Okay, so how about I change 
- the duration to 2 rounds (like the other infusion)
- add logic to apply the buff to any summoned undead
- add a condition that triggers when the caster summoned any creature and set the remaining duration to 0.1 seconds.

This way it works for spells summoning more than 1 creature, too. While also not screwing over your next turn (hopefully).

**Comment by Seinra** [comment]
It vanish!?!? The added stats should be permanent for what I know.
Ok ok, I'll be checking the Mod Finder to update your mod once you release this last version. Thanks!

**Comment by Truinto** [comment]
Yes. If the aura ends the buff gets removed from all your summons. I see no reason (in code) why the buff should persist.

**Comment by Seinra** [comment]
So they writed something in the ability description and then they did something completly different for its effect...
This is the vanilla version of the infusion:

https://pathfinderwrathoftherighteous.wiki.fextralife.com/Commanding+Infusion

"When using this infusion with animate dead, the summoned skeletons gain +4 enhancement bonus to [Strength] and [Constitution], as if affected by Augment Summoning feat."

The Augment Summoning feat adds permanent stats to the summons. Since the idea behind the Commanding Infusion is to add an "equivalent stat buff" to that granted by "augment summoning" to the next infused "animate dead" spell, I think it should also be a permanent, as if the feat is actually buffing the undead.

**Comment by Seinra** [comment]
Hello! I have to add something about this topic:

https://steamdb.info/patchnotes/13207217/

It seems Owlcat changed the infusion to give Charisma instead of Constitution, but I don't know if they also changed the way of how it works, I'll try to test the new infusion and check if it is also giving or not a permanent buff.

**Comment by Seinra** [comment]
> I will fix this

Hello! Sorry, I wasn't paying attention to the answers. I hope you can find a way to fix this bug, I'll keep an eye out for any update to Dark Codex!
